plugins/vppv2/ifplugin/descriptor/dhcp.go

```diff
@@ -202,9 +217,9 @@ func (d *DHCPDescriptor) Dump(correlate []scheduler.KVWithMetadata) ([]scheduler
 }
 
 // watchDHCPNotifications watches and processes DHCP notifications.
-func (d *DHCPDescriptor) watchDHCPNotifications(ctx context.Context, wg sync.WaitGroup, dhcpChan chan govppapi.Message) {
-	wg.Add(1)
-	defer wg.Done()
+func (d *DHCPDescriptor) watchDHCPNotifications(ctx context.Context, dhcpChan chan govppapi.Message) {
+	d.wg.Add(1)
```

Comment:
This creates race condition. `Add` is supposed to be called before goroutine is started.

Check out docs and example for more info: https://godoc.org/sync#example-WaitGroup